pandas/core/groupby/groupby.py

```diff
@@ -2076,9 +2076,8 @@ def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
                                                      limit=limit, freq=freq,
                                                      axis=axis))
         if fill_method:
-            new = copy.copy(self)
+            new = self
```

Comment:
Unfortunately doing this will change the state of self, which seems unwise. However, I also need to get this data into the obj attribute of a groupby object in order to call shift on it later. 
Is there a helper function I'm missing that will solve this issue for me?

Reply:
Don't think the branch here is necessary. If state management is the issue and the context manager outlined below works should just be able to assign to a local variable from the context manager and leave `self` untouched